### PR TITLE
fix(parsers): MiniMax M2.7 stream parity for leaked and truncated tool calls

### PR DESCRIPTION
**Pull request by Chokoyo**
#### Overview:

This PR updates MiniMax `PARSER.stream.*` parity expectations and fixes Dynamo’s stream-finalization behavior for truncated MiniMax tool-call markup.

For `PARSER.stream.2` and `PARSER.stream.3`, both vLLM and SGLang leak complete MiniMax tool-call XML into `normal_text`. Dynamo treats the completed `<minimax:tool_call>...</minimax:tool_call>` block as structured tool output and strips the protocol markup from user-visible text.

Example vLLM output for `PARSER.stream.3`:

```yaml
normal_text='<minimax:tool_call>
<invoke name="get_weather">
<parameter name="location">NYC</parameter>
</invoke>
<invoke name="get_time">
<parameter name="timezone">EST</parameter>
</invoke>
</minimax:tool_call>'
calls=[]
```

For `PARSER.stream.4.a` and `PARSER.stream.4.b`, we comply with the alignment rubric from PR #9583 by following the upstream MiniMax spec. The MiniMax M2.7 tool-calling guide defines tool calls as paired `<minimax:tool_call>...</minimax:tool_call>` XML and its parser only matches complete paired blocks:

Spec reference:
https://huggingface.co/MiniMaxAI/MiniMax-M2.7/blob/main/docs/tool_calling_guide.md

For `PARSER.stream.4.b`, Dynamo now aligns with vLLM under the parity alignment rule: both drop the incomplete in-flight call and keep protocol markup out of user-visible text. SGLang still emits an empty-argument call from an incomplete block, so that divergence is documented.

#### Details:

- Add documented MiniMax stream parity reasons for `PARSER.stream.2`, `.3`, `.4.a`, and `.4.b`.
- Fix MiniMax stream finalization so incomplete `<minimax:tool_call>` markup is not leaked into `normal_text` for `.4.a`, and `.4.b`.
- Add a regression test for MiniMax stream finalization with incomplete tool-call markup.

#### Where should the reviewer start?

- `lib/llm/src/protocols/openai/chat_completions/jail.rs`
- `lib/llm/tests/test_jail.rs`
- `tests/parity/parser/fixtures/minimax_m2/PARSER.stream.*.yaml`

#### Related Issues:
- PR #9583 for alignment rubric

## Release Notes

* **Bug Fixes**
  * Improved handling of incomplete or truncated tool-call markers by preserving leading text instead of discarding it, enhancing robustness for edge cases.

* **Tests**
  * Added regression tests for handling malformed tool-call XML and backend parsing consistency across different scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ai-dynamo/dynamo/pull/9866?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by dynamo-ops — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]
LGTM other than AI comments

File: lib/llm/src/protocols/openai/chat_completions/jail.rs [hunk 1]
```diff
@@ -1044,6 +1059,14 @@ impl JailedStream {
...
                         let content = if normal_text.as_deref() == Some("") {
                             ""
+                        } else if is_finalize
+                            && self.tool_call_parser.as_deref() == Some("minimax_m2")
+                        {
+                            // MiniMax's reference parser is strict: missing paired fences means
+                            // zero recovered calls. The raw `<minimax:tool_call>` envelope is
+                            // still protocol markup, so keep only pre-call prose at stream end.
+                            self.prefix_before_first_tool_call_marker(accumulated_content)
+                                .unwrap_or("")
```
[the remark this answers is not in the document]
Comment (Chokoyo) [reply lib/llm/src/protocols/openai/chat_completions/jail.rs]: Fixed in `58e5e60`: the MiniMax finalize branch now runs before the empty `normal_text` fallback, and I added a pre-marker prose regression case to verify the prefix is preserved while the truncated tool-call markup is dropped.

**Review by keivenchang — APPROVED** [review]
pre-approved

File: lib/llm/tests/test_jail.rs [hunk 2]
```diff
@@ -3541,6 +3541,81 @@ fahrenheit
         }
     }
 
+    /// MiniMax uses a strict reference parser: incomplete paired XML fences do
+    /// not recover a call. At stream end, the jail must still avoid surfacing
+    /// the raw `<minimax:tool_call>` protocol block as assistant content.
+    #[tokio::test]
+    async fn test_minimax_m2_stream_finalize_zero_call_truncation_drops_markup() {
```
Comment (keivenchang) [inline lib/llm/tests/test_jail.rs:3548]: thanks, can you add a comment that these are categorized as `PARSER.stream.4.a` and `PARSER.stream.4.b`?

Comment (Chokoyo) [reply lib/llm/tests/test_jail.rs:3548]: Done in `2e4b908123`: added a comment mapping the first/prefix-preservation rows to `PARSER.stream.4.a` and the mid-call truncation row to `PARSER.stream.4.b`.
